pkg/ingester/ingester_v2.go

```diff
@@ -105,6 +109,19 @@ func (u *userTSDB) PostDeletion(metrics ...labels.Labels) {
 	}
 }
 
+func (u *userTSDB) isIdle(now time.Time, idle time.Duration) bool {
+	u.lastUpdateMutex.Lock()
+	defer u.lastUpdateMutex.Unlock()
+
+	return u.lastUpdate.Add(idle).Before(now)
+}
+
+func (u *userTSDB) setLastUpdate(t time.Time) {
```

Comment:
This is called in the `v2Push()` which is quite a sensitive hot path. I'm a bit scared of the exclusive mutex. What do you think if we use an atomic storing the epoch?